src/mechglue/gss_plugin.c

```diff
@@ -117,9 +117,6 @@ gss_OID_set gss_mech_interposer(gss_OID mech_type)
         }
     }
 
-    /* while there also initiaize special_mechs */
-    (void)gpp_special_available_mechs(interposed_mechs);
```

Comment:
This initializes the static gpp_s_mechs via gpp_new_special_mech() -> gpp_add_special_oids(), wouldn't it be better to simply free the returned value to avoid the leak?

Or have you determined that this is really a complete noop because some other calls will initialize that stuff before is needed ?

Reply:
No, I just missed the code path and it didn't break anything in tests :)  Updated.